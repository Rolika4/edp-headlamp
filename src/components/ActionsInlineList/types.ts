import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
