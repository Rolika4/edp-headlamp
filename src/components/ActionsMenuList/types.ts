import { KubeObjectAction } from '../../types/actions';

export interface ActionsMenuListProps {
  actions: KubeObjectAction[];
  handleCloseActionsMenu?(): void;
  anchorEl: HTMLElement | null;
}
