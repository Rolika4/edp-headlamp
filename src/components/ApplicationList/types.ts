import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface ApplicationListProps {
    applications: EDPCodebaseKubeObjectInterface[];
    onEdit?(): void;
    onDelete?(): void;
}
