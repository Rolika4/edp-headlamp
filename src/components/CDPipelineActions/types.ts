import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineActionsProps {
    kubeObjectData: EDPCDPipelineKubeObjectInterface;
    isDetailsPage?: boolean;
}
