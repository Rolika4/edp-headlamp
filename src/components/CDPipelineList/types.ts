import { EDPCDPipelineKubeObjectInterface } from '../../k8s/EDPCDPipeline/types';

export interface CDPipelineListProps {
    CDPipelines: EDPCDPipelineKubeObjectInterface[];
}
