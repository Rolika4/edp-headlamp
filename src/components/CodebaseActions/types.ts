import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';

export interface CodebaseActionsProps {
    kubeObjectData: EDPCodebaseKubeObjectInterface;
    isDetailsPage?: boolean;
}
