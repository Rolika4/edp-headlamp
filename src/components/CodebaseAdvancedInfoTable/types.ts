import { EDPKubeObjectInterface } from '../../types/k8s';

export interface CodebaseAdvancedInfoTableProps {
    kubeObjectData: EDPKubeObjectInterface;
}
