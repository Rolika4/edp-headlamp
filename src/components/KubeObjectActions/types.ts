export interface KubeObjectAction {
    name: string;
    label: string;
    disabled?: boolean;
    action: (any) => void;
}

export interface KubeObjectActionsProps {
    actions: KubeObjectAction[];
    tooltipOpen: boolean;
    setTooltipOpen(boolean): void;
}
