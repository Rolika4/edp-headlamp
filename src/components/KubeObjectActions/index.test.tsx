/**
 * @jest-environment jsdom
 */

import { describe, test } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { createKubeAction } from '../../utils/actions/createKubeAction';
import { KubeObjectActions } from './index';
import { KubeObjectActionsProps } from './types';

describe('KubeObjectActions', () => {
  test('renders correctly', () => {
    const props: React.PropsWithChildren<KubeObjectActionsProps> = {
      actions: [
        createKubeAction({
          name: 'TestKubeAction',
          action: () => {},
          icon: 'akar-icons:plus',
        }),
        createKubeAction({
          name: 'TestDisabledKubeAction',
          action: () => {},
          icon: 'ion:rocket-outline',
          disabled: { status: true, reason: 'For testing' },
        }),
      ],
      anchorEl: document.createElement('button'),
      handleCloseActionsMenu: () => {},
    };
    render(
      <TestWrapper>
        <KubeObjectActions {...props}>
          <div>Create Action</div>
          <div>Edit Action</div>
          <div>Delete Action</div>
        </KubeObjectActions>
      </TestWrapper>
    );

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});
