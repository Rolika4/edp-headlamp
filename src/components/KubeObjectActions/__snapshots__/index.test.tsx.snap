// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KubeObjectActions renders correctly 1`] = `
<div
  class="makeStyles-actionList-2"
  role="list"
>
  <div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span />
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Testkubeaction
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div>
    <div
      aria-disabled="true"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters Mui-disabled MuiListItem-button Mui-disabled"
      role="button"
      tabindex="-1"
      title="For testing"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span />
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Testdisabledkubeaction
        </span>
      </div>
    </div>
  </div>
</div>
`;
